addons/pkg/util/tkr_util.go

```diff
@@ -90,3 +104,36 @@ func GetBootstrapPackageNameFromTKR(ctx context.Context, clt client.Client, pkgR
 
 	return "", pkgNamePrefix, fmt.Errorf("no bootstrap package prefixed with '%s' is found in the TKR object", pkgNamePrefix)
 }
+
+func FetchTKRSpecFromAnnotations(tkc map[string]string) (*runtanzuv1alpha3.TanzuKubernetesRelease, error) {
```

Comment:
you are not returning "TKRSpec", you are returning a TanzuKubernetesRlease. 
Change name of function to a more generic form. The returned object is clear from the function signature you don't need to put it in the name. 

The word "Fetch" is obvious from the signature and call to the function as well. 
something like l

func TKRFromAnnotation(tkc map[string]string) (*runtanzuv1alpha3.TanzuKubernetesRelease, error)

not sure... but definetily something less specific than FetchTKRSpecFromAnnotations


I guess following precedence of other functions in the file
func GetTKRByAnnotation 
would work too. 


and it does not seem you need to export this function (begins with upper case letter). The only place it is called from is the utils package.   It would be best to make it private, (lower case) unless there is a place where it is called from outside the package, and I don't see that in this patch.